hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -736,4 +761,25 @@ private CopyObjectResponse copyObject(String copyHeader,
       }
     }
   }
+
+  /**
+   * Parse the key and bucket name from copy header.
+   */
+  private Pair<String, String> parseSourceHeader(String copyHeader)
+      throws OS3Exception {
+    String header = copyHeader;
+    if (header.startsWith("/")) {
+      header = copyHeader.substring(1);
+    }
+    int pos = header.indexOf("/");
+    if (pos == -1) {
+      OS3Exception ex = S3ErrorTable.newError(S3ErrorTable
+          .INVALID_ARGUMENT, header);
+      ex.setErrorMessage("Copy Source must mention the source bucket and " +
+          "key: sourcebucket/sourcekey");
+      throw ex;
+    }
+
+    return Pair.of(copyHeader.substring(0, pos), copyHeader.substring(pos + 1));
```

Comment:
I think this should use header instead of copyHeader `Pair.of(header.substring(0, pos), header.substring(pos + 1))`? I think we can just use a single variable here for clarity?

Reply:
Thanks, you are right. I fixed it and created a unit test for this utility method.